server/dex/dex.go

```diff
@@ -474,6 +475,14 @@ func NewDEX(cfg *DexConf) (*DEX, error) {
 		}
 	}
 
+	// Client comms RPC server.
+	server, err := comms.NewServer(cfg.CommsCfg)
+	if err != nil {
+		abort()
+		return nil, fmt.Errorf("NewServer failed: %v", err)
+	}
+	startSubSys("Comms Server", server)
+
```

Comment:
I see how this would achieve the desired effect of shutting down the server last, but is it ok to start the server before the lower-level systems that it relies on? I realize of course that sub-system `Run` methods are run in goroutines with no guarantee of sequence anyway, but we still want ot start the server last, no?

Reply:
That's a good point about the comms order.  The issue I noticed was that messages at shutdown are dropped if comms goes does first.  But your point about start-up is taken.

Reply:
The comms subsys is moved back to its original location, but I'll let you resolve this comment in case you have thoughts about the outgoing messages on shutdown.